### PR TITLE
Use prettyunits to format `obj_size()` results

### LINKED ISSUE #57: obj_size() formatting for large objects
When I typically reach for `obj_size` I am using it on relatively large objects (MBs or GBs). I am curious if there is a recommended formatting method for converting from printing as Bytes to KB, MB, GB, etc. I can manually convert with a combination of division + `fsprint`, `paste0`, `round` etc.

There are a few `r-lib` packages that provide nice printing of bytes, but as far as I can tell do not have a method for `lobstr_bytes` objects. Does it make sense to have `prettyunits` operate natively on `lobstr_bytes` via a method or even having a function argument inside `obj_size()` for unit conversion/prettier printing?

Thank you for your time reviewing. 🖖 

``` r
library(prettyunits)
library(lobstr)
library(magrittr)

car_size <- obj_size(mtcars)

# lobstr_bytes format
class(car_size)
#> [1] "lobstr_bytes"

car_size_num <- obj_size(mtcars) %>% 
  as.numeric()

# numeric
class(car_size_num)
#> [1] "numeric"

# scales::number_bytes() is deprecated
# and only works on numeric converted obj_size
scales::number_bytes(car_size)
#> Error in UseMethod("round_any"): no applicable method for 'round_any' applied to an object of class "lobstr_bytes"
scales::number_bytes(as.numeric(car_size))
#> [1] "7 KiB"
scales::number_bytes(unclass(car_size))
#> [1] "7 KiB"

# scales::label_bytes() works for numeric or unclassed objects
# but feels a bit awkward outside of ggplot2
scales::label_bytes()(unclass(car_size))
#> [1] "7 kB"

# prettyunits::pretty_bytes works well
# but again requires numeric or unclassed conversion
prettyunits::pretty_bytes(car_size)
#> Error in as.data.frame.default(x[[i]], optional = TRUE): cannot coerce class '"lobstr_bytes"' to a data.frame
prettyunits::pretty_bytes(unclass(car_size))
#> [1] "7.21 kB"
```

<sup>Created on 2021-08-27 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

### LINKED ISSUE #60: `format()` method for `lobstr_bytes`
The carrier package used to do `format(pryr::object_size(x))`. This has stopped working when pryr started to create lobstr objects, because these don't have a format method.

### PR DESCRIPTION
**Pull request by hadley**
Fixes #57. Fixes #60.

### CODE REVIEW

File: src/size.cpp [hunk 1]
```diff
@@ -216,7 +216,7 @@ cpp11::doubles obj_csize_(cpp11::list objects, cpp11::environment base_env, int
 
   cpp11::writable::doubles out(n);
   for (int i = 0; i < n; ++i) {
-    out[i] = out[i] + obj_size_tree(objects[i], base_env, sizeof_node, sizeof_vector, seen, 0);
+    out[i] = obj_size_tree(objects[i], base_env, sizeof_node, sizeof_vector, seen, 0);
```
Comment (DavisVaughan) [inline src/size.cpp:219]: I guess this is a separate bug fix related to the relative size test? NEWS worthy?

**Review by DavisVaughan — APPROVED** [review]

File: src/size.cpp (hunk 1 above)
Comment (hadley) [reply src/size.cpp:219]: Yeah, it was adding some floating point weirdness. I don't understand why this ever worked and/or why it didn't cause more problems.

**Comment by lionel-** [comment]
Why doesn't this use the rlang bytes class?

**Comment by hadley** [comment]
@lionel- hmmmm, because `rlang::as_bytes(1e-6)` errors, but that was due to another bug that I fixed. Maybe I can reconsider

**Comment by lionel-** [comment]
But why don't we fix this in rlang?

**Comment by lionel-** [comment]
oh that was not the initial reason, got it. Should rlang use prettyunits (conditionally or inlining)? The current formatting comes from Jim's packages IIRC.

**Comment by hadley** [comment]
I think it makes sense to be consistent, and is prettyunits already designed to be inlined.

**Comment by lionel-** [comment]
Updated in https://github.com/r-lib/rlang/pull/1429.

**Pull request merged**